[docs] Document charts composition (#10710)
mui/mui-x: docs/data/charts/overview/overview.md

Inside this wrapper, you can render `<XAxis />`, `<YAxis />`, or any plot component (`<BarPlot />`, `<LinePlot />`, `<AreaPlot />`, `<ScatterPlot />`)

> To get an overview of what is available, have a look at the [composition page](/x/react-charts/composition/).

Reviewer: ```suggestion
You can check the [composition page](/x/react-charts/composition/) for more information. 
```
